### ISSUE TITLE
CI-Tests (circleci or travis)

### ISSUE THREAD

**Issue by tintinweb** [ethereum/evmlab#58]
I'd like to add travis or circleci based unittests. I can set everything up and push it to the repo we just need to aggree and configure either travis or circle for the repo.

first unittests are introduced with #57 and I expect more to come :)

cheers,
tin

**Comment by cryptomental** [comment]
Hi @tintinweb , that would indeed help a lot, especially with pull requests. Has any decision been made on this?

**Comment by tintinweb** [comment]
Hi @cryptomental,

the codebase is still undergoing quite some substantial changes and refactoring that would very often result in failing unittests and therefore require quite some effort to maintain unittests. The plan is to stabelize the interfaces first and add unittests to verify them. A decision has not yet been taken but is expected to be soon.

stay tuned :)

cheers,
tin

**Comment by holiman** [comment]
I've enabled travis now. Errored, though: https://travis-ci.org/ethereum/evmlab/builds/442581576?utm_medium=notification&utm_source=email

**Comment by holiman** [comment]
```
Installed /home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/attrdict-2.0.0-py3.6.egg
Searching for parsimonious==0.8.0
Reading https://pypi.python.org/simple/parsimonious/
Downloading https://files.pythonhosted.org/packages/4a/89/32c55944cd30dff856f16859ee325b13c83c260d0c56c0eed511e8063c87/parsimonious-0.8.0.tar.gz#sha256=ae0869d72a6e57703f24313a5f5748e73ebff836e6fe8b3ddf34ea0dc00d086b
Best match: parsimonious 0.8.0
Processing parsimonious-0.8.0.tar.gz
Writing /tmp/easy_install-wu3222jk/parsimonious-0.8.0/setup.cfg
Running parsimonious-0.8.0/setup.py -q bdist_egg --dist-dir /tmp/easy_install-wu3222jk/parsimonious-0.8.0/egg-dist-tmp-39ez1ona
  File "build/bdist.linux-x86_64/egg/parsimonious/tests/benchmarks.py", line 91
    print 'Took %.3fs to parse %.1fKB: %.0fKB/s.' % (seconds_each,
```

**Comment by tintinweb** [comment]
this is actually a parsimonious issue. [eth-abi](https://github.com/ethereum/eth-abi/blob/master/setup.py#L55) is requiring it in version 0.8.0 which had a py3 compatibility issue. that should be fixed with parsimonious 0.8.1

**Comment by tintinweb** [comment]
carver updated eth-abi to be less restrictive on the version of parsimonious fixing the py3 compatibility issue for us https://github.com/ethereum/eth-abi/commit/5342b418ad1e5cacfc9b85046d2f5ffd18bb3fa1
